### PR TITLE
basic framework auth

### LINKED ISSUE #909: Framework Authentication
Looking through the documentation, I didn't find any support for Framework authentication, which is consistent with this error:

I0219 13:00:31.122567  1734 sched.cpp:336] No credentials provided. Attempting to register without authentication
I0219 13:00:31.123420  1734 sched.cpp:1131] Got error 'Framework at scheduler-fd0ef15a-5149-4651-b61a-6f55f67cf06e@10.81.80.230:41701 is not authenticated'
I0219 13:00:31.123435  1734 sched.cpp:1941] Asked to abort the driver
I0219 13:00:31.123669  1734 sched.cpp:1177] Aborting framework 'Singularity'

Are there any plans to add this feature?

### PR DESCRIPTION
**Pull request by ssalinas**
/fixes #909 

@tpetr seemed almost too simple, does this look right to you?

### CODE REVIEW

**Comment by jhaber** [comment]
CharSet on the getBytes call?

**Comment by ssalinas** [comment]
Good call @jhaber thanks, fixed now

File: SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityDriver.java [hunk 1]
```diff
@@ -62,7 +65,15 @@
 
     this.scheduler = scheduler;
 
-    this.driver = new MesosSchedulerDriver(scheduler, frameworkInfo, configuration.getMaster());
+    if (configuration.getCredentialPrincipal().isPresent() && configuration.getCredentialSecret().isPresent()) {
+      Credential credential = Credential.newBuilder()
+        .setPrincipal(configuration.getCredentialPrincipal().get())
+        .setSecret(ByteString.copyFrom(configuration.getCredentialSecret().get().getBytes("UTF-8")))
```
Comment (jhaber) [inline SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityDriver.java]: nitpick: for Java 7+ `StandardCharsets.UTF_8` is preferred

Comment (ssalinas) [inline SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityDriver.java]: updated

**Pull request merged**